libyul/optimiser/UnusedStoreEliminator.cpp

```diff
@@ -265,15 +269,29 @@ vector<UnusedStoreEliminator::Operation> UnusedStoreEliminator::operationsFromFu
 
 void UnusedStoreEliminator::applyOperation(UnusedStoreEliminator::Operation const& _operation)
 {
-	for (auto& [statement, state]: m_stores[YulString{}])
-		if (state == State::Undecided)
+	set<Statement const*>& active =
+		_operation.location == Location::Storage ?
+		activeStorageStores() :
+		activeMemoryStores();
+
+	auto it = active.begin();
+	auto end = active.end();
+	for (; it != end;)
```

Comment:
Either
```suggestion
	for (auto it = active.begin(); it != active.end();)
```
or
```suggestion
	auto it = active.begin();
	auto end = active.end();
	while (it != end)
```
The current version is rather weird...